daemon/reload.go

```diff
@@ -312,17 +312,22 @@ func (daemon *Daemon) reloadLiveRestore(conf *config.Config, attributes map[stri
 	return nil
 }
 
-// reloadNetworkDiagnosticPort updates the network controller starting the diagnose mode if the config is valid
+// reloadNetworkDiagnosticPort updates the network controller starting the diagnostic if the config is valid
 func (daemon *Daemon) reloadNetworkDiagnosticPort(conf *config.Config, attributes map[string]string) error {
-	if conf == nil || daemon.netController == nil {
+	if daemon.netController == nil {
+		return nil
+	}
+	if conf == nil {
+		// If there is no config make sure that the diagnostic is off
+		daemon.netController.StopDiagnostic()
 		return nil
 	}
-	// Enable the network diagnose if the flag is set with a valid port withing the range
+	// Enable the network diagnostic if the flag is set with a valid port withing the range
 	if conf.IsValueSet("network-diagnostic-port") && conf.NetworkDiagnosticPort > 0 && conf.NetworkDiagnosticPort < 65536 {
 		logrus.Warnf("Calling the diagnostic start with %d", conf.NetworkDiagnosticPort)
-		daemon.netController.StartDiagnose(conf.NetworkDiagnosticPort)
+		daemon.netController.StartDiagnostic(conf.NetworkDiagnosticPort)
 	} else {
-		daemon.netController.StopDiagnose()
+		daemon.netController.StopDiagnostic()
```

Comment:
Can we move this up and return early instead of using else?

Reply:
Is this looking better?
```
func (daemon *Daemon) reloadNetworkDiagnosticPort(conf *config.Config, attributes map[string]string) error {
	if conf == nil || daemon.netController == nil || !conf.IsValueSet("network-diagnostic-port") || conf.NetworkDiagnosticPort < 1 || conf.NetworkDiagnosticPort > 65535 {
		// If there is no config make sure that the diagnostic is off
		if daemon.netController != nil {
			daemon.netController.StopDiagnostic()
		}
		return nil
	}
	// Enable the network diagnostic if the flag is set with a valid port withing the range
	logrus.Warnf("Calling the diagnostic start with %d", conf.NetworkDiagnosticPort)
	daemon.netController.StartDiagnostic(conf.NetworkDiagnosticPort)

	return nil
}
```

Reply:
Sure does. Is there some other validation on these port values?

Reply:
like?

Reply:
Like sending an error when the ports are invalid?

Reply:
I checked there is no specific validation done on the interval of the value itself, only validates that is a number, as for now. We kept it simple being a config that is only for advanced users.